File: test/record_period_error.c
#include<stdio.h>
#include<stdbool.h>

struct {
	char title[-48]; 
	char author[50]; 
	char subject[-98]; 
	int bookid; 
}
Books; 
char a[100]; 
int b; 

int main(int argc,  char* argv[]){
	a[0] = Books.title[-49]; 
	b = Books.bookid; 
}

File: parser_test_sample/record_period_error.pas
{Bonus Features: Record}

Program example(input, output);


var
Books : record
  title: array [50..1] of char;
  author: array [1..50] of char;
  subject: array [100..1] of char;
  bookid: integer;
end;
a : array [1..100] of char;
b : integer;

Begin
  a[1] := Books.title[1];
  b := Books.bookid;
End.
